tests/fixtures/config-file/cloned-config/eslintConfig.js

```diff
@@ -0,0 +1,4 @@
+let errorConfig = ["error", {}];
+module.exports = {
+    rules: { camelcase: errorConfig, "default-case": errorConfig }
```

Comment:
This test currently assumes that `camelcase` and `default-case` have different default configurations. This will most likely always be true, but should the rules be removed or their configuration options made identical, this test could pass even if the cloning behavior regresses.

Is there any way we could make custom rules and run this same test with those? That way we can guarantee that this test won’t be affected by unrelated rule changes.

Reply:
when I removed the clone statement,  the fixture which uses [rule's config from config file](https://github.com/anikethsaha/eslint/blob/fixes-12592/tests/lib/cli.js#L1164-L1175) is failing.

![image](https://user-images.githubusercontent.com/26347874/81316929-d4115580-90a9-11ea-86c7-c884c9e30434.png)


> Is there any way we could make custom rule fixtures that have different default configuration values and run this same test with those?

I do think this would be a much stronger test. by different, you meant different from the properties mentioned in the schema ?

Reply:
I was suggesting we do the same test but with fixture custom rules that are guaranteed to stay the same over time. Does that make sense?